mm2src/lp_ordermatch.rs

```diff
@@ -3549,6 +3578,110 @@ pub async fn order_status(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, St
         .map_err(|e| ERRL!("{}", e))
 }
 
+enum MakerOrderCancellationReason {
+    Fulfilled,
+    InsufficientBalance,
+    Cancelled,
+}
+enum TakerOrderCancellationReason {
+    Fulfilled,
+    ToMaker,
+    TimedOut,
+    Cancelled,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct MyOrdersFilter {
+    pub order_type: Option<String>,
+    pub initial_action: Option<String>,
+    pub base: Option<String>,
+    pub rel: Option<String>,
+    pub from_price: Option<MmNumber>,
+    pub to_price: Option<MmNumber>,
+    pub from_volume: Option<MmNumber>,
+    pub to_volume: Option<MmNumber>,
+    pub from_timestamp: Option<u64>,
+    pub to_timestamp: Option<u64>,
+    pub was_taker: Option<bool>,
+    pub status: Option<String>,
+    #[serde(default)]
+    pub include_details: bool,
+}
+
+#[derive(Deserialize, Serialize)]
+#[serde(tag = "type", content = "order")]
+enum Order {
+    Maker(MakerOrder),
+    Taker(TakerOrder),
+}
+
+impl<'a> From<&'a Order> for OrderForRpc<'a> {
+    fn from(order: &'a Order) -> OrderForRpc {
+        match order {
+            Order::Maker(o) => OrderForRpc::Maker(MakerOrderForRpc::from(o)),
+            Order::Taker(o) => OrderForRpc::Taker(TakerOrderForRpc::from(o)),
+        }
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+pub async fn orders_history_by_filter(_ctx: MmArc, _req: Json) -> Result<Response<Vec<u8>>, String> {
+    let res = json!({
+        "error": format!("'orders_history_by_filter' is only supported in native mode yet"),
+    });
+    Response::builder()
+        .status(404)
+        .body(json::to_vec(&res).expect("Serialization failed"))
+        .map_err(|e| ERRL!("{}", e))
+}
+
+/// Returns *all* uuids of swaps, which match the selected filter.
+#[cfg(not(target_arch = "wasm32"))]
+pub async fn orders_history_by_filter(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
+    use crate::mm2::database::my_orders::select_orders_by_filter;
+
+    let filter: MyOrdersFilter = try_s!(json::from_value(req));
+    let db_result = try_s!(select_orders_by_filter(&ctx.sqlite_connection(), &filter, None));
+
+    let details = if filter.include_details {
+        let mut vec = vec![];
+        for order in db_result.result.iter() {
+            let uuid = &Uuid::parse_str(order.uuid.as_str()).unwrap();
+            let order_path = my_orders_history_dir(&ctx).join(order.uuid.clone() + ".json");
+            let content = slurp(&order_path);
+            if let Ok(order) = json::from_slice::<Order>(&content) {
+                vec.push(order);
+            } else if order.order_type == *"Maker" {
+                let ordermatch_ctx = try_s!(OrdermatchContext::from_ctx(&ctx));
+                let maker_orders = ordermatch_ctx.my_maker_orders.lock().await;
+                if let Some(maker_order) = maker_orders.get(uuid) {
+                    vec.push(Order::Maker(maker_order.to_owned()));
```

Comment:
`OrderForRpc` seems a better fit since it contains more data and users will most likely expect this data to have exactly the same structure as `my_order_status`.

Reply:
@shamardy The code related to this comment has not been changed, is the current PR state final?

Reply:
It's resolved here https://github.com/KomodoPlatform/atomicDEX-API/blob/4f0710ce14389e0519e37d446460e5adf8fdf962/mm2src/lp_ordermatch.rs#L3684
I used it outside of the loop because I had problems calling `OrderForRpc::from` inside because of the different lifetimes. If you have a better way to do this please suggest it 🙂

Reply:
Let me check it one more time :slightly_smiling_face:

Reply:
Ok, let's keep it as it is, there is 1 more minor remark below in regards to this :slightly_smiling_face: